### DISCUSSION TITLE
Multiple countries

### DISCUSSION THREAD

**Discussion by mishika92** [komoot/photon#751] [General]
Hi,

I see you have a world db and also individual country extracts. In my case I need 3 or 4 countries only, and I sometimes need to add/remove countries.

Is there a way to merge multiple country extracts?

Or is there a way to convert a .pbf file (the merge of some countries) into photon db directly without the intermediate osmand db conversion that takes forever?

Thanks!

**Comment by lonvia** [comment]
There is currently no way to merge countries. You can create your own extract by importing the data into Nominatim and exporting it from there.

**Comment by falguncontractor** [comment]
hi @lonvia 
Now, do we have this feature available ?  In our case, i would like to have DBs for AU and DE and merge them . 
```
wget -O - https://download1.graphhopper.com/public/extracts/by-country-code/de/photon-db-de-latest.tar.bz2 | bzip2 -cd | tar x
wget -O - https://download1.graphhopper.com/public/extracts/by-country-code/au/photon-db-au-latest.tar.bz2 | bzip2 -cd | tar x
```

**Reply by lonvia** [reply]
https://nominatim.org/2025/08/13/photon-exports-renewed.html

**Reply by falguncontractor** [reply]
How can we only import delta from the newest downloaded db ?

**Reply by falguncontractor** [reply]
which photon version is recommended to use [latest.jsonl.zst](https://download1.graphhopper.com/public/europe/belgium/photon-dump-belgium-0.7-latest.jsonl.zst)

**Reply by falguncontractor** [reply]
I attempted to merge three large databases and run them in a Docker container. The import process took several hours before the container was fully up and running.

**Reply by falguncontractor** [reply]
hi @lonvia 
Distance Difference in Meters

> We are comparing coordinates from legacy Photon dumps and new Photon JSON dumps (Nominatim).
> For a given set of supplied addresses, we calculated the coordinate differences in meters. Below are the results:


Germany — 4,165 records

```
100m – 200m: 202 records (4.85%)
200m – 400m: 255 records (6.12%)
400m – 600m: 149 records (3.58%)
600m+: 890 records (21.37%)
```

Australia — 5,365 records

```
100m – 200m: 454 records (8.46%)
200m – 400m: 323 records (6.02%)
400m – 600m: 283 records (5.27%)
600m+: 1,155 records (21.53%)
```

Can you explain why there is huge differnce and which one is more accurate?

**Reply by mtmail** [reply]
Open a new discussion, the question is no longer about adding/merging country data. Add address examples or full query URLs, otherwise the discussion ends up in guessing.

**Comment by EPNW-Eric** [comment]
I only need a specific region of a German state, so I wrote this docker container powered by the [all-in-one `mediagis/nominatim:5.3`](https://github.com/mediagis/nominatim-docker) image for running Nominatim where I installed a `jre` that downloads the `.pbf`, uses `osmium` to cut out only the specific region, then converts it into a `jsonl` using `photon`. Even if this still needs the conversion, this might be usefull for someone since it is containerized and should easily be modifiable for any `.pbf` files:

```dockerfile
FROM mediagis/nominatim:5.3 AS builder
RUN export DEBIAN_FRONTEND=noninteractive \
    && apt-get update \
    && apt-get install -y wget apt-transport-https gpg osmium-tool \
    && wget -qO - https://packages.adoptium.net/artifactory/api/gpg/key/public \
        | gpg --dearmor \
        | tee /etc/apt/trusted.gpg.d/adoptium.gpg > /dev/null \
    && echo "deb https://packages.adoptium.net/artifactory/deb $(awk -F= '/^VERSION_CODENAME/{print$2}' /etc/os-release) main" \
        | tee /etc/apt/sources.list.d/adoptium.list \
    && apt-get update \
    && apt-get install -y temurin-21-jre \
    && rm -rf /var/lib/apt/lists/*
RUN sed -i '$ d' /app/start.sh
ENV PBF_PATH=/app/data.osm.pbf
RUN wget -O /app/photon.jar https://github.com/komoot/photon/releases/download/1.0.1/photon-1.0.1.jar
RUN wget -O niedersachsen-latest.osm.pbf https://download.geofabrik.de/europe/germany/niedersachsen-latest.osm.pbf
RUN osmium extract --bbox 9.55,51.29,10.68,51.83 niedersachsen-latest.osm.pbf -o ${PBF_PATH}
RUN echo "java -jar /app/photon.jar dump-nominatim-db -host localhost -port 5432 -database nominatim -user nominatim -password ${NOMINATIM_PASSWORD} -languages en,de -export-file /app/photon-dump.jsonl" >> /app/start.sh
RUN /app/start.sh

FROM eclipse-temurin:21-jre-noble
WORKDIR /app
COPY --from=builder /app/photon.jar /app/photon.jar
COPY --from=builder /app/photon-dump.jsonl /app/photon-dump.jsonl
RUN java -jar photon.jar import -import-file /app/photon-dump.jsonl \
	&& find /app/photon_data -name "*.lock" -delete \
	&& rm /app/photon-dump.jsonl
CMD ["java", "-jar", "photon.jar", "serve", "-default-language", "de", "-metrics-enable", "true", "-listen-ip", "0.0.0.0", "-listen-port", "2322"]
```

**Reply by lonvia** [reply]
You can skip the jsonl and directly create the Photon database. Tar it together when done, move the archive to the other container and unpack.

Or you can avoid Nominatim entirely and download and simply filter the German jsonl dump before importing:

```
zstdcat photon-dump-germany-master-latest.jsonl.zst | jq 'if .type == "Place" and .content.[0].address?.state? != "Niedersachsen" then empty end' | java -jar photon.jar import -import-file -
```

**Reply by EPNW-Eric** [reply]
I would lose the bounding box extraction if I filter just based on state, so I think I need the Nominatim step:
```dockerfile
RUN osmium extract --bbox 9.55,51.29,10.68,51.83 niedersachsen-latest.osm.pbf -o ${PBF_PATH}
```
However, as you suggested, I skipped the jsonl export and copy the database over directly now:
```dockerfile
FROM mediagis/nominatim:5.3 AS builder
RUN export DEBIAN_FRONTEND=noninteractive \
    && apt-get update \
    && apt-get install -y wget apt-transport-https gpg osmium-tool \
    && wget -qO - https://packages.adoptium.net/artifactory/api/gpg/key/public \
        | gpg --dearmor \
        | tee /etc/apt/trusted.gpg.d/adoptium.gpg > /dev/null \
    && echo "deb https://packages.adoptium.net/artifactory/deb $(awk -F= '/^VERSION_CODENAME/{print$2}' /etc/os-release) main" \
        | tee /etc/apt/sources.list.d/adoptium.list \
    && apt-get update \
    && apt-get install -y temurin-21-jre \
    && rm -rf /var/lib/apt/lists/*
RUN sed -i '$ d' /app/start.sh
ENV PBF_PATH=/app/data.osm.pbf
RUN wget -O /app/photon.jar https://github.com/komoot/photon/releases/download/1.1.0/photon-1.1.0.jar
RUN wget -O niedersachsen-latest.osm.pbf https://download.geofabrik.de/europe/germany/niedersachsen-latest.osm.pbf
RUN osmium extract --bbox 9.55,51.29,10.68,51.83 niedersachsen-latest.osm.pbf -o ${PBF_PATH}
RUN echo "java -jar /app/photon.jar import -host localhost -port 5432 -database nominatim -user nominatim -password ${NOMINATIM_PASSWORD} -data-dir /app/photon_data -languages en,de" >> /app/start.sh
RUN /app/start.sh
RUN find /app/photon_data -name "*.lock" -delete

FROM eclipse-temurin:21-jre-noble
WORKDIR /app
COPY --from=builder /app/photon.jar /app/photon_data /app/
CMD ["java", "-jar", "photon.jar", "serve", "-default-language", "de", "-metrics-enable", "true", "-listen-ip", "0.0.0.0", "-listen-port", "2322"]
```

For some reason, I still need to delete the lockfiles after the import is done... Any idea why?
